app/src/main/java/org/fossasia/openevent/app/data/repository/EventRepository.java

```diff
@@ -0,0 +1,105 @@
+import org.fossasia.openevent.app.data.models.Event_Table;
+import org.fossasia.openevent.app.data.models.User;
+import org.fossasia.openevent.app.data.network.EventService;
+import org.fossasia.openevent.app.data.repository.contract.IEventRepository;
+
+import javax.inject.Inject;
+
+import io.reactivex.Observable;
+import timber.log.Timber;
+
+public class EventRepository extends Repository implements IEventRepository {
+
+    @Inject
+    public EventRepository(IUtilModel utilModel, IDatabaseRepository databaseRepository, EventService eventService) {
+        super(utilModel, databaseRepository, eventService);
+    }
+
+    @Override
+    public Observable<User> getOrganiser(boolean reload) {
+        Observable<User> diskObservable = Observable.defer(() ->
+            databaseRepository
+                .getAllItems(User.class)
+                .firstElement()
+                .toObservable()
+        );
+
+        Observable<User> networkObservable = Observable.defer(() ->
+            eventService
+                .getUser(getAuthorization())
+                .doOnNext(user -> databaseRepository
+                    .save(User.class, user)
+                    .subscribe()
+                )
+        );
+
+        return new AbstractObservableBuilder<User>(utilModel)
+            .reload(reload)
+            .withDiskObservable(diskObservable)
+            .withNetworkObservable(networkObservable)
+            .build();
+    }
+
+    @Override
+    public Observable<Event> getEvent(long eventId, boolean reload) {
+        Observable<Event> diskObservable = Observable.defer(() ->
+            databaseRepository
+                .getItems(Event.class, Event_Table.id.eq(eventId))
+                .filter(Event::isComplete)
+                .take(1)
+        );
+
+        Observable<Event> networkObservable = Observable.defer(() ->
+            eventService
+                .getEvent(eventId)
+                .doOnNext(event -> {
+                    event.setComplete(true);
+                    databaseRepository
+                        .save(Event.class, event)
+                        .subscribe();
+                })
+        );
+
+        return new AbstractObservableBuilder<Event>(utilModel)
+            .reload(reload)
+            .withDiskObservable(diskObservable)
+            .withNetworkObservable(networkObservable)
+            .build();
+    }
+
+    @Override
+    public Observable<Event> getEvents(boolean reload) {
+        Observable<Event> diskObservable = Observable.defer(() ->
+            databaseRepository.getAllItems(Event.class)
+        );
+
+        Observable<Event> networkObservable = Observable.defer(() ->
+            eventService.getEvents(getAuthorization())
+                .doOnNext(events -> databaseRepository.saveList(Event.class, events)
+                    .subscribe())
+                .flatMapIterable(events -> events))
+                .doOnEach(eventNotification -> {
```

Comment:
Because this is eventNotification now, doOnNext will skip the error one and we don't want. doOnEach and doOnNext are almost same, and this was added in a PR long ago, I had made a well-informed decision, but can't remember why. But this was preferred in this case